backend/substrapp/utils.py

```diff
@@ -186,3 +193,16 @@ def get_remote_file_content(url, auth, content_hash, salt=None):
     if computed_hash != content_hash:
         raise NodeError(f"url {url}: hash doesn't match {content_hash} vs {computed_hash}")
     return response.content
+
+
+def get_and_put_remote_file_content(url, auth, content_hash, salt=None, content_dst_path=None):
```

Comment:
The signature seems incorrect, when using stream=True, content_dst_path should be set. The following signature seems better: 
```python
def get_and_put_remote_file_content(url, auth, content_hash, content_dst_path, salt=None):
```

Same comment for the function `get_and_put_asset_content`.